Client/Frontend/Browser/TabScrollController.swift

```diff
@@ -245,8 +245,16 @@ private extension TabScrollingController {
     }
 
     func animateToolbarsWithOffsets(_ animated: Bool, duration: TimeInterval, headerOffset: CGFloat, footerOffset: CGFloat, alpha: CGFloat, completion: ((_ finished: Bool) -> Void)?) {
+        guard let scrollView = scrollView else { return }
+        let initialContentOffset = scrollView.contentOffset
+
+        // If this function is used to fully animate the toolbar from hidden to shown, keep the page from scrolling by adjusting contentOffset.
+        let isShowFromScratch = self.headerTopOffset == -UIConstants.TopToolbarHeight && headerOffset == 0
```

Comment:
comparing the header offset to `topScrollHeight` might be better. For example the height of the toptoolbar on the iPad is larger than `TopToolbarHeight` because it also includes the toptabs bar.

Reply:
Nice catch!